nixos/modules/services/audio/mympd.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.services.mympd;
+in {
+  options = {
+
+    services.mympd = {
+
+      enable = lib.mkEnableOption (lib.mdDoc "MyMPD server");
+
+      package = lib.mkPackageOption pkgs "mympd" {};
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        description = lib.mdDoc ''
+          The HTTP port where mympd's web interface will be available.
+
+          The HTTPS/SSL port can be configured via {option}`config`.
+        '';
+        example = "8080";
+      };
+
+      ssl = lib.mkOption {
+        type = lib.types.bool;
+        description = lib.mdDoc ''
+          Whether to enable listening on the SSL port.
+
+          Refer to <https://jcorporation.github.io/myMPD/configuration/configuration-files#ssl-options>
+          for more information.
+        '';
+        default = false;
+      };
+
+      config = lib.mkOption {
+        type = with lib.types; attrsOf (nullOr (oneOf [ str bool int ]));
+        description = lib.mdDoc ''
+          Manages the configuration files declaratively. For all the configuration
+          options, see <https://jcorporation.github.io/myMPD/configuration/configuration-files>.
+
+          Each key represents the "File" column from the upstream configuration table, and the
+          value is the content of that file.
+
+          The configuration is automatically written at service start-up, so it is
+          possible to change their values after the first start of the service.
+        '';
+      };
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    services.mympd.config = {
+      http_port = cfg.port;
+      ssl = cfg.ssl;
+    };
```

Comment:
Why do you expose the `port` and `ssl` options, if they are treated as part of the `config` anyway?

Reply:
BTW it'd be nice to also have a `openFirewall` option, as implemented in:

https://github.com/NixOS/nixpkgs/blob/4fa61810859ac4bbcdf0e8a1014040fac21a83c7/nixos/modules/programs/weylus.nix#L12-L35

Reply:
@doronbehar , I've chosen to expose `port` because I consider it the single most important option, all the rest are optional. For `ssl`, I've did it in order to disable it by default.

I've added an `openFirewall` option as well.

Reply:
If you move the port option into the proposed RFC42 `settings` then it will show up on [search.nixos.org](https://search.nixos.org).  examples: 

* https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=services.xonotic
* https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=services.grafana.settings.server.http_port

Reply:
@pbsds , done.